src/definition.cpp

```diff
@@ -878,6 +878,7 @@ bool readCodeFragment(const char *fileName,
     }
   }
   result = transcodeCharacterStringToUTF8(result);
+  if (result.right(1) != "\n") result += "\n";
```

Comment:
`result.right(1)` is pretty inefficient for lange strings. Better use something like `!result.isEmpty() && result.at(result.length()-1)!='\n'`